usermods/battery_status_basic/usermod_v2_battery_status_basic.h

```diff
@@ -0,0 +1,260 @@
+#pragma once
+
+#include "wled.h"
+
+
+
+
+// pin defaults
+#ifndef TEMPERATURE_PIN
+  #ifdef ARDUINO_ARCH_ESP32
+    #define BATTERY_MEASUREMENT_PIN A0
+  #else //ESP8266 boards
+    #define BATTERY_MEASUREMENT_PIN A0
+  #endif
+#endif
+
+
+// the frequency to check the battery, 1 minute
+#ifndef USERMOD_BATTERY_MEASUREMENT_INTERVAL
+  #define USERMOD_BATTERY_MEASUREMENT_INTERVAL 30000
+#endif
+
+
+// default for 18650 battery
+// https://batterybro.com/blogs/18650-wholesale-battery-reviews/18852515-when-to-recycle-18650-batteries-and-how-to-start-a-collection-center-in-your-vape-shop
+// Discharge voltage: 2.5 volt + .1 for personal safety
+#ifndef USERMOD_BATTERY_MIN_VOLTAGE
+  #define USERMOD_BATTERY_MIN_VOLTAGE 2.6
+#endif
+
+#ifndef USERMOD_BATTERY_MAX_VOLTAGE
+  #define USERMOD_BATTERY_MAX_VOLTAGE 4.2
+#endif
+
+class UsermodBatteryBasic : public Usermod {
+  private:
+    // battery pin can be defined in my_config.h
+    int8_t batteryPin = BATTERY_MEASUREMENT_PIN;
+    // how often to read the battery voltage
+    unsigned long readingInterval = USERMOD_BATTERY_MEASUREMENT_INTERVAL;
+    unsigned long lastTime = 0;
+    // battery min. voltage
+    float minBatteryVoltage = USERMOD_BATTERY_MIN_VOLTAGE;
+    // battery max. voltage
+    float maxBatteryVoltage = USERMOD_BATTERY_MAX_VOLTAGE;
+    float rawValue = 0.0;
+    // calculated voltage            
+    float voltage = 0.0;
+    // mapped battery level based on voltage
+    long batteryLevel = 0;
+    
+    /*
+    // set your config variables to their boot default value (this can also be done in readFromConfig() or a constructor if you prefer)
+    bool testBool = false;
+    unsigned long testULong = 42424242;
+    float testFloat = 42.42;
+    String testString = "Forty-Two";
+
+    // These config variables have defaults set inside readFromConfig()
+    int testInt;
+    long testLong;
+    int8_t testPins[2];
+    */
+
+  public:
+    //Functions called by WLED
+
+    // https://forum.arduino.cc/t/floating-point-using-map-function/348113/2
+    double mapf(double x, double in_min, double in_max, double out_min, double out_max) {
+      return (x - in_min) * (out_max - out_min) / (in_max - in_min) + out_min;
+    }
+
+    /*
+     * setup() is called once at boot. WiFi is not yet connected at this point.
+     * You can use it to initialize variables, sensors or similar.
+     */
+    void setup() {
+      pinMode(batteryPin, INPUT);
```

Comment:
```
bool PinManagerClass::isPinOk(byte gpio, bool output)
{
  if (gpio <  6) return  true;
  if (gpio < 12) return false; //SPI flash pins
  
  #ifdef ESP8266
  if (gpio < 17) return true;
  #else //ESP32
  if (gpio < 34) return true;
  if (gpio < 40 && !output) return true; //34-39 input only
  #endif
  
  return false;
}
```

i tried that... but it doesn't work due to this method.. 
A0 would be GPIO 17 on an esp8266

![gpio17](https://user-images.githubusercontent.com/45098772/129493237-b704b2d4-e6ad-40f4-bdc8-569112ae5aca.PNG)

Reply:
A0 is not a GPIO and cannot be _reserved_ as such.
Still it can conflict with analog button (unfortunately 8266 only has a single analog pin).

My suggestion is to ignore pin reservation if building for 8266 but use it for ESP32.